node/node.go

```diff
@@ -118,6 +119,25 @@ func New(cfg *Config, version string) (*Node, error) { //nolint:gocyclo,funlen
 	services := make([]service.Service, 0)
 
 	chain := blockchain.New(database, cfg.Network, log)
+
+	// Verify that cfg.Network is compatible with the database.
+	height, err := chain.Height()
+	if err != nil && !errors.Is(err, db.ErrKeyNotFound) {
+		return nil, fmt.Errorf("get chain height from database: %v", err)
+	}
+	if height > 0 {
+		var block *core.Block
+		block, err = chain.BlockByNumber(height)
+		if err != nil {
+			return nil, fmt.Errorf("get latest block from database: %v", err)
+		}
+		// We assume there is at least one transaction in the blockchain, otherwise VerifyBlockHash
+		// will succeed for any network with only post-0.7 blocks using the standard fallback sequencer address.
```

Comment:
> will succeed for any network with only post-0.7 blocks

Is that correct? post-0.7 blocks all have ChainID in their hash.

Reply:
Pre-0.7 blocks [have](https://github.com/NethermindEth/juno/blob/27c1ac20c794d8c7c95c71daa0669415b628669f/core/block.go#L183) the chain id in their hash, post-0.7 blocks [do not](https://github.com/NethermindEth/juno/blob/27c1ac20c794d8c7c95c71daa0669415b628669f/core/block.go#L214-L231).